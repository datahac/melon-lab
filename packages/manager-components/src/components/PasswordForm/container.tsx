import { withFormik } from 'formik';
import { compose, withHandlers } from 'recompose';
import * as Yup from 'yup';

import PasswordForm from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    password: Yup.string().required('Password is required.'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values, 'passwordForm');
    }
  },
});

const withFormHandler = compose(
  withHandlers({
    onChange: props => (values, event) => {
      props.setFieldValue(event.target.name, values.value);
    },
  }),
);

export default compose(
  withFormValidation,
  withFormHandler,
)(PasswordForm);
