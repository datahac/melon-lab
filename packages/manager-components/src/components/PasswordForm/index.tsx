import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';

import styles from './styles.css';

interface FormValues {
  password: string;
}

export interface PasswordFormProps {
  values: FormValues;
  handleSubmit: () => void;
  handleReset: () => void;
  onCancel: () => void;
  handleBlur?: () => void;
  onChange?: () => void;
  touched?: any;
  errors?: any;
}

export const PasswordForm: StatelessComponent<PasswordFormProps> = ({
  values,
  handleBlur,
  onChange,
  touched,
  errors,
  handleSubmit,
  onCancel,
  handleReset,
}) => {
  const handleCancel = () => {
    handleReset();
    onCancel();
  };

  return (
    <div className="password-form">
      <style jsx>{styles}</style>
      <Form>
        <div className="password-form__input">
          <Input
            value={values.password}
            type="password"
            placeholder="Password"
            name="password"
            onChange={onChange}
            onBlur={handleBlur}
            required={true}
            error={touched.password && errors.password}
          />
        </div>
        <div className="password-form__buttons">
          <div className="password-form__button">
            <Button type="button" style="secondary" onClick={handleCancel}>
              Cancel
            </Button>
          </div>
          <div className="password-form__button">
            <Button type="submit" onClick={handleSubmit}>
              Confirm
            </Button>
          </div>
        </div>
      </Form>
    </div>
  );
};

export default PasswordForm;
