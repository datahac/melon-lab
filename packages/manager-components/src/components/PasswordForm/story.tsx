import { storiesOf } from '@storybook/react';
import React from 'react';
import PasswordForm from './container';

const initialProps = {
  initialValues: {
    password: '',
  },
};

storiesOf('Components|Password Form', module).add('Default', () => {
  return <PasswordForm {...initialProps} />;
});
