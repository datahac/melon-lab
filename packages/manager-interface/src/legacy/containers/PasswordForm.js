import { connect } from 'react-redux';
import PasswordForm from '@melonproject/manager-components/components/PasswordForm/container';
import { actions } from '../actions/modal';

const mapStateToProps = state => ({
  initialValues: {
    password: '',
  },
});

const mapDispatchToProps = dispatch => ({
  onSubmit: values => {
    dispatch(actions.passwordEntered(values.password));
  },
  onCancel: () => {
    dispatch(actions.cancel());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PasswordForm);
