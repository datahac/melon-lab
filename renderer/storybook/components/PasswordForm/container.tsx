import { withFormik } from 'formik';
import * as Yup from 'yup';

import PasswordForm from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    password: Yup.string()
      .required('Password is required.')
      .min(8, 'Password needs to be at least 8 chars long. For your security!'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values, 'passwordForm'),
});

export default withFormValidation(PasswordForm);
