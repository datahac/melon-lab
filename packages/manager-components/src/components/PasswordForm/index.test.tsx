import React from 'react';
import PasswordForm from './container';

const initialProps = {
  initialValues: {
    password: '',
  },
};

describe('PasswordForm', () => {
  const defaultElement = <PasswordForm {...initialProps} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
