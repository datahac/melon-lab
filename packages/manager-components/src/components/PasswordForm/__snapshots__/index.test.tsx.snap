// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForm should render correctly 1`] = `
<Formik
  enableReinitialize={true}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "password": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
